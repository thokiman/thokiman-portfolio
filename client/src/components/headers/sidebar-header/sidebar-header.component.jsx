import React from "react";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import { createStructuredSelector } from "reselect";
import { selectAboutRoute } from "../../../redux/about/about.selectors";
import { selectPortfolioRoute } from "../../../redux/collection/collection.selectors";
import { selectContactRoute } from "../../../redux/contact/contact.selectors";
import { selectIsSideBarHidden } from "../../../redux/header/header.selectors";
import { selectServiceRoute } from "../../../redux/service/service.selectors";
import {
  SideBarHeaderContainer, SideBarHeaderLink, SideBarHeaderTextContainer
} from "./sidebar-header.styles";


export const SideBarHeader = ({
  isSideBarHidden,
  location: { pathname },
  aboutRoute,
  portfolioRoute,
  serviceRoute,
  contactRoute,
}) => {
  const homeText = "Home";
  const AboutText = "About";
  const portfolioText = "Portfolio";
  const serviceText = "Service";
  const contactText = "Contact";
  return (
    <SideBarHeaderContainer $issidebarhidden={isSideBarHidden}>
      <SideBarHeaderTextContainer>
        <SideBarHeaderLink $matchpath={!!pathname.match(RegExp(/^\/$/))} to="/">
          {homeText}
        </SideBarHeaderLink>
        <SideBarHeaderLink
          to={aboutRoute}
          $matchpath={!!pathname.match(RegExp(`${ aboutRoute }+`))}
        >
          {AboutText}
        </SideBarHeaderLink>

        <SideBarHeaderLink
          to={portfolioRoute}
          $matchpath={!!pathname.match(RegExp(`${ portfolioRoute }+`))}
        >
          {portfolioText}
        </SideBarHeaderLink>
        <SideBarHeaderLink
          to={serviceRoute}
          $matchpath={!!pathname.match(RegExp(`${ serviceRoute }+`))}
        >
          {serviceText}
        </SideBarHeaderLink>
        <SideBarHeaderLink
          to={contactRoute}
          $matchpath={pathname.match(RegExp(`${ contactRoute }+`))}
        >
          {contactText}
        </SideBarHeaderLink>
      </SideBarHeaderTextContainer>
    </SideBarHeaderContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  isSideBarHidden: selectIsSideBarHidden,
  aboutRoute: selectAboutRoute,
  portfolioRoute: selectPortfolioRoute,
  serviceRoute: selectServiceRoute,
  contactRoute: selectContactRoute,
});

export default withRouter(connect(mapStateToProps)(SideBarHeader));
