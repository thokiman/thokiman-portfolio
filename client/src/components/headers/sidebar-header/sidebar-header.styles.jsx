import { Link } from 'react-router-dom';
import styled, { css } from 'styled-components';

const maxWidthSmartphone = 800;
const greenBlueColor = '#128194';
const whiteSmokeColor = '#e3e1e4';

const SideBarContainerIsActivated = css`
  left: 0;
  transition: all 0.5s ease;
  z-index: 2;
  opacity: 1;
  transition: all 0.5s ease;
`;

const getSideBarContainer = ({ $issidebaractive }) => {
  if ($issidebaractive) return SideBarContainerIsActivated;
  return null;
};

export const SideBarHeaderContainer = styled.div`
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    width: 140px;
    position: absolute;
    top: 14vh;
    left: 140px;
    bottom: 0;
    transition: all 0.5s ease;
    display: flex;
    align-items: flex-start;
    overflow-y: hidden;
    opacity: 0;
    ${getSideBarContainer}
  }
`;

export const SideBarHeaderTextContainer = styled.div`
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    width: 100%;
    position: relative;
    top: 10vh;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: flex-start;
  }
`;

const SideBarHeaderTextActive = css`
  border-bottom: ${greenBlueColor} solid 1px;
  color: ${whiteSmokeColor};
  transition: all 0.5s linear;
`;
const SideBarHeaderTextHide = css`
  border: none;
  color: ${greenBlueColor};
  transition: all 0.5s linear;
`;
const getSideBarHeaderTextActive = ({ $matchpath }) => {
  if ($matchpath) return SideBarHeaderTextActive;
  return SideBarHeaderTextHide;
};
export const SideBarHeaderLink = styled(Link)`
  display: none;
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    ${getSideBarHeaderTextActive}

    text-decoration: none;
    width: fit-content;
    height: 100%;
    display: block;
    text-align: center;
    padding: 0 4px;
    font-size: 30px;
    margin: 10px 0;
    padding: 5px 0;
  }
`;
