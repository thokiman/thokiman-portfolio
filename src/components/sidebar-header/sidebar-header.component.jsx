import React from "react";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { withRouter } from "react-router-dom";

import {
  SideBarHeaderContainer,
  SideBarHeaderTextContainer,
  SideBarHeaderLink,
} from "./sidebar-header.styles";
import { selectIsSideBarHidden } from "../../redux/header/header.selectors";
import { selectAboutRoute } from "../../redux/about/about.selectors";
import { selectPortfolioRoute } from "../../redux/collection/collection.selectors";
import { selectServiceRoute } from "../../redux/service/service.selectors";
import { selectContactRoute } from "../../redux/contact/contact.selectors";

const SideBarHeader = ({
  isSideBarHidden,
  location: { pathname },
  aboutRoute,
  portfolioRoute,
  serviceRoute,
  contactRoute,
}) => {
  return (
    <SideBarHeaderContainer $issidebarhidden={isSideBarHidden}>
      <SideBarHeaderTextContainer>
        <SideBarHeaderLink $matchpath={!!pathname.match(RegExp(/^\/$/))} to="/">
          Home
        </SideBarHeaderLink>

        <SideBarHeaderLink
          to={aboutRoute}
          $matchpath={!!pathname.match(RegExp(`${aboutRoute}+`))}
        >
          About
        </SideBarHeaderLink>

        <SideBarHeaderLink
          to={portfolioRoute}
          $matchpath={!!pathname.match(RegExp(`${portfolioRoute}+`))}
        >
          Portfolio
        </SideBarHeaderLink>

        <SideBarHeaderLink
          to={serviceRoute}
          $matchpath={!!pathname.match(RegExp(`${serviceRoute}+`))}
        >
          Service
        </SideBarHeaderLink>

        <SideBarHeaderLink
          to={contactRoute}
          $matchpath={pathname.match(RegExp(`${contactRoute}+`))}
        >
          Contact
        </SideBarHeaderLink>
      </SideBarHeaderTextContainer>
    </SideBarHeaderContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  isSideBarHidden: selectIsSideBarHidden,
  aboutRoute: selectAboutRoute,
  portfolioRoute: selectPortfolioRoute,
  serviceRoute: selectServiceRoute,
  contactRoute: selectContactRoute,
});

export default withRouter(connect(mapStateToProps)(SideBarHeader));
