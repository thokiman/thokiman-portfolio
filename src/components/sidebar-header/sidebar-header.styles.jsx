import { Link } from "react-router-dom";
import styled, { css } from "styled-components";

const SideBarContainerHidden = css`
  left: 0;
  transition: all 0.5s ease;
`;

const getSideBarContainer = (props) => {
  if (props.$issidebarhidden) return SideBarContainerHidden;
  return null;
};

export const SideBarHeaderContainer = styled.div`
  @media only screen and (max-width: 800px) {
    width: 140px;
    position: absolute;
    top: 14vh;
    left: 140;
    bottom: 0;
    transition: all 0.5s ease;
    display: flex;
    align-items: flex-start;
    overflow-y: hidden;
    ${getSideBarContainer}
  }
`;

export const SideBarHeaderTextContainer = styled.div`
  @media only screen and (max-width: 800px) {
    width: 100%;
    position: relative;
    top: 50px;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: flex-start;
  }
`;

const SideBarHeaderTextActive = css`
  border-bottom: #128194 solid 1px;
  color: #e3e1e4;
  transition: all 0.5s linear;
`;
const SideBarHeaderTextHide = css`
  border: none;
  color: #128194;
  transition: all 0.5s linear;
`;
const getSideBarHeaderTextActive = (props) => {
  if (props.$matchpath) return SideBarHeaderTextActive;
  return SideBarHeaderTextHide;
};
export const SideBarHeaderLink = styled(Link)`
  display: none;
  @media only screen and (max-width: 800px) {
    ${getSideBarHeaderTextActive}
    text-decoration: none;
    width:fit-content;
    height: 100%;
    display: block;
    text-align: center;
    padding: 0 4px;
    font-size: 30px;
    margin: 10px 0;
    padding: 5px 0;
    
    
`;
