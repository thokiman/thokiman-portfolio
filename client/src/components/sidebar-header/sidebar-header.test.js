import React from "react";
import { shallow } from "enzyme";

import { SideBarHeader } from "./sidebar-header.component";

describe("SideBarHeader component", () => {
  let wrapper;
  const mockRoute = "/";

  const mockFalseBool = false;

  beforeEach(() => {
    const mockProps = {
      isSideBarHidden: mockFalseBool,
      location: { pathname: mockRoute },
      aboutRoute: mockRoute,
      portfolioRoute: mockRoute,
      serviceRoute: mockRoute,
      contactRoute: mockRoute,
    };
    wrapper = shallow(<SideBarHeader {...mockProps} />);
  });

  it("should render SideBarHeader component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
