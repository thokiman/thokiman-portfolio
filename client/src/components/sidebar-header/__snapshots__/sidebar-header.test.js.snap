// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBarHeader component should render SideBarHeader component 1`] = `ShallowWrapper {}`;
